uniffi_bindgen/src/bindings/python/gen_python/visit_mut.rs

```diff
@@ -0,0 +1,569 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use anyhow::{bail, Result};
+
+use heck::{ToShoutySnakeCase, ToSnakeCase, ToUpperCamelCase};
+use once_cell::sync::Lazy;
+
+use std::collections::{BTreeSet, HashSet};
+use std::fmt::Debug;
+
+use super::Config;
+use crate::interface::ir::*;
+
+// Taken from Python's `keyword.py` module.
+static KEYWORDS: Lazy<HashSet<String>> = Lazy::new(|| {
+    let kwlist = vec![
+        "False",
+        "None",
+        "True",
+        "__peg_parser__",
+        "and",
+        "as",
+        "assert",
+        "async",
+        "await",
+        "break",
+        "class",
+        "continue",
+        "def",
+        "del",
+        "elif",
+        "else",
+        "except",
+        "finally",
+        "for",
+        "from",
+        "global",
+        "if",
+        "import",
+        "in",
+        "is",
+        "lambda",
+        "nonlocal",
+        "not",
+        "or",
+        "pass",
+        "raise",
+        "return",
+        "try",
+        "while",
+        "with",
+        "yield",
+    ];
+    HashSet::from_iter(kwlist.into_iter().map(|s| s.to_string()))
+});
+
+/// Implements `VisitMut` to specialize the BindingsIr for Python
+pub struct BindingsIrVisitor {
+    pub config: Config,
+    // The following fields are populated as we walk the tree for VisitMut
+    pub imports: BTreeSet<String>,
+    pub protocols: Vec<Protocol>,
+    pub runtimes: Runtimes,
+    pub exports: Vec<String>,
+}
```

Comment:
This module essentially replaces the `CodeOracle` system (as well as a couple others).  Is it better to have all of the code in one place or try to split it out over multiple modules?  I think I like it all in one place, but I could be convinced to try to split it up.

One thing I definitely like over the old system is that there's a lot less indirection involved: `BindingsIrVisitor` sets data, then the filters read it and that's basically the whole store.